app/models/rule_event.rb

```diff
@@ -0,0 +1,29 @@
+class RuleEvent
+  def self.load(attributes)
+    object = allocate
+    object.instance_variable_set(:@timestamp, DateTime.parse(attributes.fetch("timestamp")))
+    object.instance_variable_set(:@details, attributes.fetch("details").map { RuleEventDetail.load(_1) })
+    object
+  end
+
+  def initialize
+    @timestamp = DateTime.now.utc
```

Comment:
What purpose does this timestamp have?

Reply:
UI purposes. Otherwise you know "last event was X" but can't know if this one dates 1 minute or 10 days ago.